web/server/codechecker_server/api/report_server.py

```diff
@@ -2829,54 +2830,89 @@ def massStoreRun(self, name, tag, version, b64zip, force,
                     # Round the duration to seconds.
                     durations = int(sum(check_durations))
 
-                # This session's transaction buffer stores the actual run data
-                # into the database.
-                with DBSession(self.__Session) as session:
-                    # Load the lock record for "FOR UPDATE" so that the
-                    # transaction that handles the run's store operations
-                    # has a lock on the database row itself.
-                    run_lock = session.query(RunLock) \
-                        .filter(RunLock.name == name) \
-                        .with_for_update(nowait=True).one()
-
-                    # Do not remove this seemingly dummy print, we need to make
-                    # sure that the execution of the SQL statement is not
-                    # optimised away and the fetched row is not garbage
-                    # collected.
-                    LOG.debug("Storing into run '%s' locked at '%s'.",
-                              name, run_lock.locked_at)
-
-                    # Actual store operation begins here.
-                    user_name = self.__get_username()
-                    run_id = store_handler.addCheckerRun(session,
-                                                         command,
-                                                         name,
-                                                         tag,
-                                                         user_name,
-                                                         run_history_time,
-                                                         version,
-                                                         force,
-                                                         cc_version,
-                                                         statistics)
-
-                    self.__store_reports(session,
-                                         report_dir,
-                                         source_root,
-                                         run_id,
-                                         file_path_to_id,
-                                         run_history_time,
-                                         self.__context.severity_map,
-                                         wrong_src_code_comments,
-                                         skip_handler,
-                                         checkers)
-
-                    store_handler.setRunDuration(session,
+                # When we use multiple server instances and we try to run
+                # multiple storage to each server which contain at least two
```

Comment:
Please extend the comment here with the information that the same report means the same report hash and that is the key for the review data table and both of the store actions try to update the same review data row.